src/Application.vala

```diff
@@ -0,0 +1,60 @@
+/*-
+ * Copyright 2018-2023 elementary, Inc. (https://elementary.io)
+ *           2014 Marvin Beckers <beckersmarvin@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Marvin Beckers <beckersmarvin@gmail.com>
+ */
+
+public class PantheonCalculator.Application : Gtk.Application {
+    private MainWindow window;
+
+    construct {
+        application_id = "io.elementary.calculator";
+        flags = ApplicationFlags.FLAGS_NONE;
+
+        Intl.setlocale (LocaleCategory.ALL, "");
+        Intl.bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+        Intl.bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+        Intl.textdomain (GETTEXT_PACKAGE);
+    }
+
+    public override void startup () {
+        base.startup ();
+
+        var quit_action = new SimpleAction ("quit", null);
+        var undo_action = new SimpleAction ("undo", null);
```

Comment:
I don't think those should be application actions. At least, the undo action should be an window action using the `Gtk.Widget.add_shortcut()` method, an `Gtk.Shortcut` using an `Gtk.KeyvalTrigger` and `Gtk.NamedAction` should replace the `set_accels_for_action()` method.

Also, Gtk.Window already provides an `window.close` action that better represent the wanted effect, since it will close only the focused window instead of all open windows.

Reply:
Yeah I think I agree that `undo` should be a window action, I'll fix that. I think `quit` should be an app action that closes all windows though since there's already window manager shortcuts for closing individual windows